auction-app-frontend/src/components/FeaturedProducts/FeaturedProducts.jsx

```diff
@@ -0,0 +1,78 @@
+
+import { useState, useEffect } from "react";
+
+import { ProductCard, LoadingComponent, ErrorComponent } from "src/components";
+
+import { findFeaturedProducts, findPopularProducts } from "src/services";
+import { useUser } from "src/store/UserContext";
+
+import "./style.scss";
+
+const FeaturedProducts = () => {
+  const [products, setProducts] = useState([]);
+  const [loading, setLoading] = useState();
+  const [error, setError] = useState(null);
+
+  const { userId } = useUser();
+
+  const fetchFeaturedProducts = ()  => {
+    setLoading(true);
+
+    findFeaturedProducts(userId)
+      .then((response) => {
+        setProducts(response);
+        setLoading(false);
+      })
+      .catch((error) => {
+        setError(error);
+        setLoading(false);
+      }).finally(() => {
+        setLoading(false);
+      });
+  }
+
+  const fetchPopularProducts = ()  => {
+    setLoading(true);
+
+    findPopularProducts()
+      .then((response) => {
+        setProducts(response);
+        setLoading(false);
```

Comment:
If we have finally that will happen at the end of async call, and we set there loading to false, do we need it here and in catch block?

Reply:
we dont need it in the then block, because finally will set it to false anyways, i will have to remove this from other places too